lib/instrumentation/@elastic/elasticsearch.js

```diff
@@ -0,0 +1,85 @@
+/*
+ * Copyright 2023 New Relic Corporation. All rights reserved.
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+'use strict'
+
+module.exports = function initialize(_agent, elastic, _moduleName, shim) {
+  shim.setDatastore(shim.ELASTICSEARCH)
+  shim.setParser(queryParser)
+
+  shim.recordQuery(elastic.Transport.prototype, 'request', function wrapQuery(shim, _, __, args) {
+    const ctx = this
+    return {
+      query: JSON.stringify(args?.[0]),
+      promise: true,
+      opaque: true,
+      inContext: function inContext() {
+        getConnection.call(ctx, shim)
+      }
+    }
+  })
+}
+
+function queryParser(params) {
+  params = JSON.parse(params)
+  const path = params.path.split('/')
+  const actions = {
+    GET: 'search',
+    PUT: 'create',
+    POST: 'create',
+    DELETE: 'delete',
+    HEAD: 'exists'
+  }
+  const defaultCollection = 'any'
+  let operation
+  let collection
+
+  const suffix = actions[params.method]
+  path.forEach((segment, idx) => {
+    const prev = idx - 1
```

Comment:
I worry that someone could create an index as `_search` or name an index with `_`.  In that case we're naming the collection any. It appears that's not allowed but we aren't handling that at all

Reply:
Yeah, it's explicitly prohibited for path index names to start with `_`, `-`, or `+`, and for path parameters to include `/`, among other punctuation characters. https://www.elastic.co/guide/en/elasticsearch/reference/current/indices-create-index.html

Reply:
OK--I've updated that to handle index creation separately.